src/main/java/net/jodah/failsafe/internal/util/CircularBitSet.java

```diff
@@ -22,8 +22,24 @@ public CircularBitSet(int size, CircularBitSet oldBitSet) {
 
     // Initialize from oldBitSet
     if (oldBitSet != null) {
-      for (int i = 0; i < size && i < oldBitSet.occupiedBits; i++)
+      initializeFromOldBits(oldBitSet);
+    }
+  }
+
+  /**
+   * Initialize bitSet from oldBitSet with correct order
+   */
+  private void initializeFromOldBits(CircularBitSet oldBitSet) {
```

Comment:
After `bs2` is created, the value should be `{ false, true }`. With that change, the rest of the test makes sense:

```
    CircularBitSet bs2 = new CircularBitSet(2, bs); // expected { false, true }
    assertEquals(bs2.occupiedBits(), 2);
    assertEquals(bs2.positiveRatio(), .5);
    assertEquals(bs2.negativeRatio(), .5);

    bs2.setNext(true); // expected { true, true }
    assertEquals(bs2.occupiedBits(), 2);
    assertEquals(bs2.positiveRatio(), 1.0);
    assertEquals(bs2.negativeRatio(), 0.0);
```

It might be easier to debug this stuff if BitSet had a better toString implementation. Maybe we should add one to CircularBitSet that shows an array representation of the BitSet, something like:

```
  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append('[');
    for (int i = 0; i < bitSet.length(); i++) {
      if (i > 0)
        sb.append(',');
      sb.append(bitSet.get(i) ? 1 : 0);
    }
    return sb.append(']').toString();
  }
```